cibuildwheel/options.py

```diff
@@ -775,6 +775,14 @@ def _compute_build_options(self, identifier: str | None) -> BuildOptions:
                         # default to manylinux2014
                         image = pinned_images["manylinux2014"]
                     elif config_value in pinned_images:
+                        if config_value in {
+                            "manylinux1",
+                            "manylinux2010",
+                            "manylinux_2_24",
+                            "musllinux_1_1",
+                        }:
+                            msg = f"Unmaintained image, either upgrade to a supported image or pin directly to {pinned_images[config_value]!r}"
```

Comment:
Adding a bit more detail...

```suggestion
                            msg = f"Deprecated image {config_value!r}. This value will not work in a future version of cibuildwheel. Either upgrade to a supported image or continue using the deprecated image by pinning directly to {pinned_images[config_value]!r}."
```